lms/djangoapps/courseware/grades.py

```diff
@@ -30,6 +31,48 @@
 log = logging.getLogger("edx.courseware")
 
 
+EXCLUDED_LOCATION_CATEGORIES = {'video', 'html', 'discussion'}
+def descriptor_filter(descriptor):
+    """
+    Pass a location filter to the field data cache so that we don't
+    instantiate video descriptors during grading
+    """
+    return descriptor.location.category not in EXCLUDED_LOCATION_CATEGORIES
+
+
+class MaxScoresCache(object):
+    """docstring for MaxScoresCache"""
+    def __init__(self, locations, cache=cache):
+        self.locations = locations
+        self._max_scores_cache = {}
+        self._max_scores_updates = {}
+
+    def fetch_from_cache(self):
+        self._max_scores_cache = cache.get_many(
+            [unicode(loc) for loc in self.locations]
+        )
+
+    def push_to_cache(self):
+        if self._max_scores_updates:
+            cache.set_many(self._max_scores_updates, 180)  # 3 mins
+
+    def add_max_score(self, location, max_score):
```

Comment:
Again with naming, since it's a `MaxScoresCache`, maybe we could just call it get/set instead of add_max_score?